shared/logutil/logutil.go

```diff
@@ -28,3 +30,28 @@ func ConfigurePersistentLogging(logFileName string) error {
 	logrus.Info("File logging initialized")
 	return nil
 }
+
+//Masks the url credentials before logging for security purpose
+//[scheme:][//[userinfo@]host][/]path[?query][#fragment] -->  [scheme:][//[***]host][/***][#***]
+//if the format is not matched nothing is done, string is returned as is.
+func MaskCredentialsLogging( currUrl string) string {
+	//error if the input is not a URL
```

Comment:
```suggestion
```
This comment doesn't add any additional context to the code. 
A reader can understand that an input URL that fails to parse would return an error without this comment.